enterprise/frontend/src/embedding-sdk/components/private/SdkThemeProvider.tsx

```diff
@@ -0,0 +1,32 @@
+import { useMemo } from "react";
+
+import type { MetabaseTheme } from "embedding-sdk";
+import {
+  getEmbeddingThemeOverride,
+  setGlobalEmbeddingColors,
+} from "embedding-sdk/lib/theme";
+import { useSelector } from "metabase/lib/redux";
+import { getSettings } from "metabase/selectors/settings";
+import { ThemeProvider } from "metabase/ui";
+import { getApplicationColors } from "metabase-enterprise/settings/selectors";
+
+interface Props {
+  theme?: MetabaseTheme;
+  children: React.ReactNode;
+}
+
+export const SdkThemeProvider = ({ theme, children }: Props) => {
+  const appColors = useSelector(state =>
+    getApplicationColors(getSettings(state)),
+  );
```

Comment:
The extra component is needed because of the provider structure, the call to `useSelector` needs to be below `ReduxProvider` but above `ThemeProvider`